enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,215 @@
+
+```go
+// Release represents an OpenShift release in the release graph.
+// +k8s:deepcopy-gen=true
+type Release struct {
+  // version is a semantic versioning identifying the update version. When this
+  // field is part of spec, version is optional if image is specified.
+  Version string `json:"version"`
+
+  // image is a container image location that contains the update. When this
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // metadata is additional metadata associated with the release, such
+  // as associated channels (under
+  // io.openshift.upgrades.graph.release.channels) or a landing page
+  // (under url).
+  //
+  // +optional
+  Metadata map[string]string `json:"metadata,omitempty"`
+}
+```
+
+and to use that type instead of the current `Update` for `ClusterVersionStatus` properties.
+
+Then, in the cluster-version operator (CVO), to port existing logic like [available-update translation][cluster-version-operator-update-translation] and [available-update lookup][cluster-version-operator-update-lookup] to preserve the Cincinnati metadata.
+In some cases (when an administrator requested an update that was not in the available update set), this would require an additional Cincinnati request to retrieve metadata about the requested release.
+
+### B
+
+Similar to [*a*](#a), except instead of changing the `ClusterVersionStatus` fields, we'd deprecate them and make `Release` a stand-alone custom resource.
+`Version` would become `ObjectMeta.Name`, and `Image` and `Metadata` would be `Status` properties.
+Then we need some way to figure out which is the current release and which are the available updates, which is a bit racier now that we can no longer retrieve a single, atomic snapshot with a `GET ClusterVersion`.
+
+### i
+
+We'd also add some additional properties:
+
+```go
+// Source a the release version for which updates to this release can be launched.
+// For example, source on a 4.1.1 release might be "4.1.0".
+Source string `json:"source"`
+```
+
+In addition, the current release (being reconciled by the cluster-version operator) would have a `release.openshift.io/role` annotation with a `current` value.
+
+### ii
+
+We'd add new `ClusterVersionStatus` properties like:
+
+```
+// Current is the release version which is currently being reconciled by the cluster-version operator.
+Current string `json:"current,omitempty"`
+
+// AvailableUpdatesV2 holds release versions to which the current version may be updated.
+AvailableUpdatesV2 []string `json:"availableUpdatesV2,omitempty`
+```
+
+### iii
+
+Leave the current `ClusterVersionStatus` alone and just continue to fill the now-redundant `Update` properties (like `Image`) and use their `Version` properties instead of [*ii*](#ii)'s bare strings.
+
+### Metadata properties
+
+This enhancement declares the following to be well-known metadata properties:
```

Comment:
Let's make it clear that these properties are part of clusterversion API and not just some well known constants.

because the cluster version API would be responsible for maintaining the compatibility even if cincy changes these keys.

Reply:
I've pushed wording about tying this to `apiVersion` with 182442f -> 8d8a080.  How does that look to you?